app/demo-interactor/KNDemoKernel.cu

```diff
@@ -162,6 +163,10 @@ void initialize(const CudaGridParams&  grid,
                 const StateDeviceRef&  states,
                 const InitialPointers& initial)
 {
+    static const KernelParamCalculator calc_kernel_params(
+        initialize_kernel, "initialize", grid.block_size);
+    calc_kernel_params(states.size());
```

Comment:
```suggestion
    auto params = calc_kernel_params(states.size());
```
and use params in the kernel launch?  [Or remove if it is not used?]

Reply:
Right now `params` isn't used because there's a grid-based stride inside the demo kernel. I'm going to remove that as part of the performance testing (fewer registers when it's removed). In the meantime though, calling `calc_kernel_params` registers the state size and the kernel launch count, so it's not a null-op because of the diagnostics side effect.